### PR TITLE
Add client prep test

### PR DESCRIPTION
**Pull request by spuiuk**
The ansible scripts to build the setup the client machine and enable testing.

This at the moment pulls the tests from spuiuk/samba-integration:tests. Once we have the tests branch in gluster/samba-integration, we will have to update the scripts to use those instead and can be merged into the gluster/samba-integration:master branch.

### CODE REVIEW

File: vagrant/ansible/roles/client.test/tasks/main.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+---
+- name: Fetch the latest samba-integration-tests repo
+  git:
+    repo: 'https://github.com/spuiuk/samba-integration.git'
+    version: tests
+    dest: /root/samba-integration-tests
```
Comment (obnoxxx) [inline vagrant/ansible/roles/client.test/tasks/main.yml:6]: shouldn't use `/root`

File: vagrant/ansible/roles/client.test/tasks/main.yml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+- name: Create a symlink for test-info.yml file
+  file:
+    src: /root/test-info.yml
+    dest: /root/samba-integration-tests/test-info.yml
```
Comment (obnoxxx) [inline vagrant/ansible/roles/client.test/tasks/main.yml:11]: `/root`

File: vagrant/ansible/roles/client.test/tasks/main.yml [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    owner: root
+    group: root
+    state: link
+
+- name: Run tests
+  command:
+    chdir: /root/samba-integration-tests
```
Comment (obnoxxx) [inline vagrant/ansible/roles/client.test/tasks/main.yml]: `/root`

File: vagrant/Makefile [hunk 4]
```diff
@@ -21,6 +21,9 @@ setup.test.only:
 setup.cluster.only:
 	@$(VSSH) setup "make -C /home/vagrant/ansible setup.cluster"
 
-setup.cluster: setup.prep setup.test.only setup.cluster.only
+client1.test:
+	@$(VSSH) client1 "sudo make -C /root/samba-integration-tests tests"
```
Comment (obnoxxx) [inline vagrant/Makefile:25]: I am not 100% sure if this is clean: I thought we should trigger the tests from the setup machine. Just like we do the setting up of the cluster and client machine from the setup machine.

File: vagrant/ansible/roles/client.test/tasks/main.yml [hunk 5]
```diff
@@ -0,0 +1,19 @@
+---
+- name: Fetch the latest samba-integration-tests repo
+  git:
+    repo: 'https://github.com/spuiuk/samba-integration.git'
```
Comment (obnoxxx) [inline vagrant/ansible/roles/client.test/tasks/main.yml:4]: We should not merge with using this git repo.
We now have the tests branch. And once there's something there that we can run as test, we can put the final repo here before merging.

File: vagrant/ansible/roles/node.prep/tasks/main.yml [hunk 6]
```diff
@@ -39,24 +39,3 @@
...
-    - name: modify copr repo to only use it for compat-gnutls packages
-      lineinfile:
-        dest: /etc/yum.repos.d/_copr_sergiomb-SambaAD.repo
-        line: "includepkgs=compat-gnutls34.* compat-nettle32.*"
-        insertafter: '\[copr:copr.fedorainfracloud.org:sergiomb:SambaAD\]'
-
-  when: ansible_distribution == 'CentOS' and ansible_distribution_major_version == '7'
-
-- name: Enable Samba nightly rpms repository
-  command: yum-config-manager --add-repo http://artifacts.ci.centos.org/gluster/nightly-samba/samba-nightly-master.repo
```
Comment (anoopcs9) [inline vagrant/ansible/roles/node.prep/tasks/main.yml:62]: We can also move
```
- name: Enable GlusterFS nightly rpms repository
  command: yum-config-manager --add-repo http://artifacts.ci.centos.org/gluster/nightly/master.repo
```
to _common.prep_.

Comment (obnoxxx) [reply vagrant/ansible/roles/node.prep/tasks/main.yml:62]: why do we need the gluster rpms on the client machine?

File: vagrant/Makefile (hunk 4 above)
Comment (obnoxxx) [reply vagrant/Makefile:25]: Also `/root` has to go...

File: vagrant/ansible/roles/client.test/tasks/main.yml (hunk 5 above)
Comment (obnoxxx) [reply vagrant/ansible/roles/client.test/tasks/main.yml:4]: we can now use `https://github.com/gluster/samba-integration.git` as PR #50 is merged.

File: vagrant/ansible/roles/client.test/tasks/main.yml [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+    version: tests
+    dest: /root/samba-integration-tests
+
+- name: Create a symlink for test-info.yml file
+  file:
+    src: /root/test-info.yml
+    dest: /root/samba-integration-tests/test-info.yml
+    owner: root
```
Comment (obnoxxx) [inline vagrant/ansible/roles/client.test/tasks/main.yml:12]: root

File: vagrant/ansible/roles/client.test/tasks/main.yml (hunk 1 above)
Comment (spuiuk) [reply vagrant/ansible/roles/client.test/tasks/main.yml:6]: This is within the client vm which is created by scratch and the tests have to be run as root. I don't understand why we cannot use absolute paths here.

File: vagrant/ansible/roles/node.prep/tasks/main.yml (hunk 6 above)
Comment (anoopcs9) [reply vagrant/ansible/roles/node.prep/tasks/main.yml:62]: > why do we need the gluster rpms on the client machine?

oops..
please ignore.

File: vagrant/Makefile (hunk 4 above)
Comment (spuiuk) [reply vagrant/Makefile:25]: The last patch takes the calling off from client.test(where it was finally called through setup) and calls it through the make file instead. Makes it a bit easier to test for developers to trigger the tests by themselves on the client without ssh-ing into the client or the setup.

File: vagrant/ansible/roles/client.test/tasks/main.yml (hunk 5 above)
Comment (spuiuk) [reply vagrant/ansible/roles/client.test/tasks/main.yml:4]: Yes. The PR was labelled as do not merge for this reason. I put it up so Anoop and Gunther could test it.

File: vagrant/ansible/roles/node.prep/tasks/main.yml (hunk 6 above)
Comment (spuiuk) [reply vagrant/ansible/roles/node.prep/tasks/main.yml:62]: We allow mounting using a fuse-mount if required by the tests. This was enabled for use by gluster-fuse.

File: vagrant/Makefile (hunk 4 above)
Comment (obnoxxx) [inline vagrant/Makefile:25]: Now that #50 is merged, and has a changed target name, rename it here:
```suggestion
	@$(VSSH) client1 "sudo make -C /root/samba-integration-tests test"
```

**Comment by spuiuk** [comment]
New patches uploaded. This is however dependent on patches in #56.

Still left have the label "Do not merge" since we depend on patches in #56.

**Comment by obnoxxx** [comment]
#56 merged - this needs to be rebased

**Comment by spuiuk** [comment]
From the console output. Both the prep and the tasks were run.
16:29:30 
16:29:30 TASK [client.test.prep : Run tests] ********************************************
16:29:42 changed: [client1]
16:29:42 
16:29:42 TASK [client.test : Run tests] *************************************************
16:29:54 changed: [client1]

**Pull request merged**